modules/http4s/src/main/scala/natchez/http4s/NatchezMiddleware.scala

```diff
@@ -93,9 +93,42 @@ object NatchezMiddleware {
    * - "client.http.status_code" -> "200", "403", etc. // why is this a string?
    *
    */
-  def client[F[_]: Trace](client: Client[F])(
-    implicit ev: MonadCancel[F, Throwable]
-  ): Client[F] =
+  def client[F[_] : Trace : MonadCancelThrow](client: Client[F]): Client[F] =
+    NatchezMiddleware.client(client, _ => Seq.empty)
+
+  /**
+   * A middleware that adds the current span's kernel to outgoing requests, performs requests in
+   * a span called `http4s-client-request`, and adds the following fields to that span.
+   *
+   * - "client.http.method"      -> "GET", "PUT", etc.
+   * - "client.http.uri"         -> request URI
+   * - "client.http.status_code" -> "200", "403", etc. // why is this a string?
+   *
+   * @param client the `Client[F]` to be enhanced
+   * @param additionalAttributes additional attributes to be added to the span
+   * @tparam F An effect with instances of `Trace[F]` and `MonadCancelThrow[F]`
+   * @return the enhanced `Client[F]`
+   */
+  def clientWithAttributes[F[_] : Trace : MonadCancelThrow](client: Client[F])
+                                                           (additionalAttributes: (String, TraceValue)*): Client[F] =
+    NatchezMiddleware.client(client, (_: Request[F]) => additionalAttributes)
+
+  /**
+   * A middleware that adds the current span's kernel to outgoing requests, performs requests in
+   * a span called `http4s-client-request`, and adds the following fields to that span.
+   *
+   * - "client.http.method"      -> "GET", "PUT", etc.
+   * - "client.http.uri"         -> request URI
+   * - "client.http.status_code" -> "200", "403", etc. // why is this a string?
+   *
+   * @param client the `Client[F]` to be enhanced
+   * @param additionalAttributes a function that takes the `Request[F]` and returns any additional attributes to be added to the span
+   * @tparam F An effect with instances of `Trace[F]` and `MonadCancelThrow[F]`
+   * @return the enhanced `Client[F]`
+   */
+  def client[F[_] : Trace : MonadCancelThrow](client: Client[F],
+                                              additionalAttributes: Request[F] => Seq[(String, TraceValue)],
```

Comment:
Thoughts on accepting an effectual method here?

```suggestion
                                              additionalAttributes: Request[F] => F[Seq[(String, TraceValue)]],
```

Reply:
Yes, that's a great idea. I'll update the PR accordingly.

Reply:
@armanbilge This has been updated if you want to take another look.